.github/workflows/release.yml

```
@@ -63,3 +63,178 @@ jobs:
         run: python -u ./dev/releases/make_github_release.py ${GITHUB_REF#refs/tags/} tmp/
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  cygwin:
+    name: "Create Windows ${{ matrix.arch }} installer"
+    needs: unix
+    runs-on: windows-latest
+    # TODO Do we want to skip these jobs for pull requests?  If so, add the following line, and delete the PR-specific code from the steps below.
+    # if: ${{ github.event_name != 'pull_request' }}
```

```suggestion
      if: ${{ github.event_name != 'pull_request' }}
```
I'd say let's not run this on every pull request. It takes 1h 20mins compared to the CI tests' 50 mins. Maybe instead add a `workflow_dispatch:` to the triggers of the workflow? Then whenever a PR touches the release system, we could manually trigger the full workflow, that is including the windows installer jobs. (In theory, we could also let the full workflow be triggered automatically on every PR with an `topic: infrastructure` label.)

---

Agreed, and I prefer the manual trigger over label magic (we don't use labels very consistently)

---

As @ChrisJefferson suggested, we can also run this once a week. Of course it would be nice to have slack notifications so we don't miss when something breaks..

---

I agree that there should be a `on: workflow_dispatch:` trigger (indeed I think all of our workflows should have that trigger). I think the same about having `on: schedule`  triggers too (although there we have to decide the frequency, and which branch(es) should be tested on any regular schedule).

I will respond to the other topic(s) in a separate comment in the main thread.